rust/datafusion/src/physical_plan/hash_join.rs

```diff
@@ -186,8 +190,18 @@ impl ExecutionPlan for HashJoinExec {
                         })
                         .await?;
 
+                    let num_rows: usize =
+                        left_data.1.iter().map(|batch| batch.num_rows()).sum();
```

Comment:
Maybe we want to put this behind a flag so we don't do this in the macro so we don't compute the `num_rows` when not being logged?